doc/readDaq.md

```diff
@@ -0,0 +1,151 @@
+Only need this if the user uses the NiFpga_AcquireFifoReadElementsU16() function above.
+
+The NiFpga_AcquireFifoReadElementsU16() function will transfer FIFO’s memory to the user, and it is needed to release the elements from this memory region because it belongs to FIFO.
+The NiFpga_ReleaseFifoElements() function “releases previously acquired FIFO elements” according to the NiFpga_ReleaseFifoElements() [function description](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/functions_fifo_method.html), and it says that “the host must release elements after acquiring them”.
+
+You are not supposed to use the NiFpga_ReleaseFifoElements()] function after the NiFpga_ReadFifoU16() function, because the NiFpga_ReadFifoU16() function allocates memory region by the user, therefore, the elements will not be in the FIFO’s memory when the NiFpga_ReleaseFifoElements() function tries to release those elements.
+If this is done, you will get the [error -61073](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/errors.html).
+
+Here is an example to release elements from the `daqFIFO` using C/C++
+
+```c 
+size_t elements = 0;
+
+NiFpga_ReleaseFifoElements(session, NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO, elements);
+```
+
+where:
+- `session` handles a currently FPGA session
+- `NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO` is the FIFO from which to release
+- `elements` is the number of elements to release
+
+## FIFO is full in the FPGA side
+
+The maximum number of elements the `Target-to-host DMA` FIFO can hold is `2^M - 1`, where M = 16 (address width).
+For example, if you want to transfer 5 elements using the FIFO, you are still able to add 10 elements more before FIFO is full (assuming there is no reader).
+This is because the maximun number of elements is 15 (2^16 - 1).
+If you want to transfer 23 elements using FIFO, the maximun number of elements now is 31, because LabVIEW coerces the maximun number of elements to the clossest one less than a power of two value.
+
+The [Write FIFO Method](https://www.ni.com/docs/en-US/bundle/labview-fpga-module/page/lvfpga/fifo_write.html) provides you the `Timed Out?` terminal.
+This is helpfull to know whether the FIFO is full or not.
+
+Continuing with the first example, if you want to transfer 5 elements, after adding the 16th element to FIFO, the `Timed Out?` terminal will be True.
+This `Timed Out?` terminal will be wired to the `daqFifoFull` indicator.
+
+## daqFifoFull indicator
+
+The `daqFifoFull` is a boolean indicator.
+If `daqFifo` is full, `daqFifoFull` will be True and data is not written to the FIFO until space becomes available.
+
+The C/C++ developer needs to take actions to avoid FIFO being full and losing data.
+If the overflow happens, the FPGA side should provide a way (TBD) to compensate this.
+This is not part of this document, since we are just trying to organize the results here instead of design.
```

Comment:
If the overflow happens, the FPGA side should provide a way to compensate this.
The details is TBD.

Remove the sentence of: "This is not part of this document, since we are just trying to organize the results here instead of design."

Reply:
Done.